src/components/user/unavailability/unavailability.pg.repository.ts

```diff
@@ -0,0 +1,97 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  ID,
+  NotFoundException,
+  ServerException,
+  Session,
+  UnsecuredDto,
+} from '../../../common';
+import { DtoRepository, Pg } from '../../../core';
+import { paginate, sorting } from '../../../core/database/query';
+import {
+  CreateUnavailability,
+  Unavailability,
+  UnavailabilityListInput,
+} from './dto';
+
+@Injectable()
+export class UnavailabilityPgRepository extends DtoRepository(Unavailability) {
+  @Inject(Pg) private readonly pg: Pg;
+
+  async create(input: CreateUnavailability, _session: Session) {
+    const rows = await this.pg.query<{ id: ID }>(
+      `
+      INSERT INTO sc.person_unavailabilities(period_start, period_end, description, 
+        admin_people_id, created_by_admin_people_id, modified_by_admin_people_id, 
+        owning_person_admin_people_id, owning_group_admin_groups_id)
+      VALUES($1, $2, $3, $4, 
+          (SELECT admin_people_id FROM admin.tokens WHERE token = 'public'), 
+          (SELECT admin_people_id FROM admin.tokens WHERE token = 'public'), 
+          (SELECT admin_people_id FROM admin.tokens WHERE token = 'public'), 
+          (SELECT id FROM admin.groups WHERE  name = 'Administrators'))
+      RETURNING id;
+      `,
+      [input.start, input.end, input.description, input.userId]
+    );
+    if (!rows) {
+      throw new ServerException('Failed to create unavailability');
+    }
+    return rows[0].id;
+  }
+
+  async readOne(id: ID): Promise<UnsecuredDto<Unavailability>> {
+    const rows = await this.pg.query<UnsecuredDto<Unavailability>>(
+      `
+      SELECT id, admin_people_id, period_start, period_end, description, 
```

Comment:
According to the `Unavailability` dto we only need `description`, `start` and `end` fields. Please omit the next columns: `admin_people_id`, `created_by_admin_people_id as "creator"`, `modified_at as "modifiedAt"`, `modified_by_admin_people_id`

Reply:
just pushed commit [e3bc2fb] that removes the unnecessary fields